edu.wpi.first.wpilib.plugins.riolog/src/netconsole3/RioConsole.java

```diff
@@ -0,0 +1,325 @@
+  private Consumer<Boolean> connectedCallback = null;
+
+  @Override
+  protected void finalize() {
+    stop();
+  }
+
+  public void stop() {
+    cleanup.set(true);
+    closeSocket();
+    if (listener != null) {
+      listener.interrupt();
+    }
+  }
+
+  public void reconnect() {
+    reconnect.set(true);
+    closeSocket();
+  }
+
+  public boolean getAutoReconnect() {
+    return autoReconnect;
+  }
+
+  public void setAutoReconnect(boolean value) {
+    lock.lock();
+    try {
+      autoReconnect = value;
+      if (value) {
+        wakeupListener.signal();
+      }
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  public boolean getPaused() {
+    return paused.get();
+  }
+
+  public boolean setPaused(boolean value) {
+    return paused.getAndSet(value);
+  }
+
+  public boolean getDiscard() {
+    return discard.get();
+  }
+
+  public boolean setDiscard(boolean value) {
+    return discard.getAndSet(value);
+  }
+
+  public boolean getShowWarning() {
+    return showWarning.get();
+  }
+
+  public boolean setShowWarning(boolean value) {
+    return showWarning.getAndSet(value);
+  }
+
+  public boolean getShowPrint() {
+    return showPrint.get();
+  }
+
+  public boolean setShowPrint(boolean value) {
+    return showPrint.getAndSet(value);
+  }
+
+  public void putDummyMessage() {
+    try {
+      messageQueue.put(new DummyMessage());
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+    }
+  }
+
+  private void setSocket(Socket socket) {
+    lock.lock();
+    this.socket = socket;
+    lock.unlock();
+  }
+
+  public void setConnectedCallback(Consumer<Boolean> callback) {
+    lock.lock();
+    this.connectedCallback = callback;
+    lock.unlock();
+  }
+
+  public BlockingQueue<Message> getMessageQueue() {
+    return messageQueue;
+  }
+
+  private static byte[] emptyFrame = new byte[] {0, 0};
+
+  private DataInputStream connect(int team) {
+    logger.log("connecting to team " + team);
+    Socket mySocket;
+    try {
+      mySocket = rioConnector.connect(team);
+    } catch (InterruptedException e) {
+      return null;
+    }
+    if (mySocket == null) {
+      return null;
+    }
+    setSocket(mySocket);
+    DataInputStream in;
+    OutputStream out;
+    try {
+      mySocket.setTcpNoDelay(true);  // for keep alives
+      in = new DataInputStream(mySocket.getInputStream());
+      out = mySocket.getOutputStream();
+    } catch (IOException e) {
+      closeSocket();
+      return null;
+    }
+    lock.lock();
+    Consumer<Boolean> connCb = connectedCallback;
+    lock.unlock();
+    if (connCb != null) {
+      connCb.accept(Boolean.TRUE);
+    }
+    //logger.log("socket connected");
+
+    // kick off keep alive thread
+    sender = new Thread(() -> {
+      while (!Thread.interrupted() && !cleanup.get()) {
+        try {
+          Thread.currentThread().sleep(2);
+          out.write(emptyFrame);
+          out.flush();
+        } catch (InterruptedException e) {
```

Comment:
Reset the thread's interrupted flag, see above.

Reply:
I'm not seeing a reason to do that here.  We're in the main thread function, there is no caller.